ipc/observability/src/tracing_layers.rs

```diff
@@ -0,0 +1,109 @@
+// Copyright 2022-2024 Protocol Labs
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use std::ops::Deref;
+use tracing::{field::Visit, Event, Subscriber};
+use tracing_subscriber::Registry;
+use tracing_subscriber::{layer::Context, Layer};
+
+const DOMAIN_FIELD: &str = "domain";
+const EVENT_FIELD: &str = "event";
+
+pub struct DomainEventFilterLayer<L> {
+    domains: Option<Vec<String>>,
+    events: Option<Vec<String>>,
+    inner: L,
+}
+
+impl Deref for DomainEventFilterLayer<Registry> {
+    type Target = Registry;
+
+    fn deref(&self) -> &Self::Target {
+        &self.inner
+    }
+}
+
+impl<L> DomainEventFilterLayer<L> {
+    pub fn new(domains: Option<Vec<String>>, events: Option<Vec<String>>, inner: L) -> Self {
+        DomainEventFilterLayer {
+            domains,
+            events,
+            inner,
+        }
+    }
+}
+
+impl<S, L> Layer<S> for DomainEventFilterLayer<L>
+where
+    S: Subscriber,
+    L: Layer<S>,
+{
+    fn on_event(&self, event: &Event, ctx: Context<S>) {
+        if self.domains.is_none() && self.events.is_none() {
```

Comment:
It seems like using `match (self.domain, self.events)` can simplify your downstream processing

Reply:
I am not sure, would I not need to then match for all variats (none, none), (some, none), (none, some), (some, some) and essentially repeat the same filtering code all over?